src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,89 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class SalaryInfo {
+    private Map<String, Integer> nameSalaryMap = new LinkedHashMap<>();
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
+        if (names == null || data == null || dateFrom == null || dateTo == null) {
+            return null;
+        }
+
+        Map<String, Integer> resultInfoMap = new LinkedHashMap<>();
```

Comment:
I've seen your comment about Map usage, but the task is to do it manually. If do not have ideas how to implement it we can provide you support in Slack or you can go to Q&A with this question.
Also, there are 5 loops in your solution - we need only 2.
![image](https://github.com/mate-academy/jv-salary-info/assets/20326858/0ac12dc1-91fc-4aa8-b689-426662b70a3a)

Reply:
you still have Map - delete it